### PR TITLE
Les 94  Integrate login page with BE

### PR DESCRIPTION
**Pull request by Kurilo-Pavel**

### CODE REVIEW

**Review by valer-sky — APPROVED** [review]

File: src/pages/LoginPage.tsx [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+                htmlFor='remember'
+                className='labelCheckbox'
+              >
+                Stay logged in
+              </label>
+            </div>
+            <Button
+              buttonType={'submit'}
+              buttonText={'Sign in'}
+              onClick={undefined}
```
Comment (Terapool) [inline src/pages/LoginPage.tsx:92]: onClick={undefined} - bad approuch. See comment in Button component file. Just do not pass this property

File: src/pages/LoginPage.tsx [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+interface FormValues {
+  email: string;
+  password: string;
+  remember: boolean;
+}
+
+const SignupSchema = Yup.object().shape({
+  email: Yup.string()
+    .matches(mailRgex)
+    .min(3, 'Email should be more than 3 characters')
```
Comment (Terapool) [inline src/pages/LoginPage.tsx:26]: please use constants for such 3 / 256 etc.

https://jeffreyeverhart.com/2020/09/28/javascript-tips-using-named-constants-for-magic-numbers/

File: src/components/PasswordAndConfirm.tsx [hunk 3]
```diff
@@ -40,10 +41,16 @@ const PasswordAndConfirm = ({minSymbol, maxSymbol, isConfirm}: PasswordProps): J
     <div className='password'>
       <label className='passwordLabel'>{isConfirm ? 'Confirm password' : 'Password'}
         <input type={visiblePassword ? 'text' : 'password'}
-               className={'passwordInput ' + `${(error && isBlur) ? 'errorInput' : ''}`}
-               onChange={fieldHandler}
+               minLength={minSymbol}
+               maxLength={maxSymbol}
+               value={value}
+               placeholder='At least 6 characters'
```
Comment (Terapool) [inline src/components/PasswordAndConfirm.tsx:47]: placeholder={`At least ${minSymbol} characters`}

File: src/components/Button.tsx [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import "./button.scss"
+import {MouseEventHandler} from "react";
+
+type ButtonProps = {
+  buttonType: 'button' | 'submit' | 'reset' | undefined;
+  buttonText: string;
+  onClick: MouseEventHandler <HTMLButtonElement> | undefined;
```
Comment (Terapool) [inline src/components/Button.tsx:7]: just mark onClick as not required property, no need to pass here undefined in explicit form:

onClick?: ...

**Review by Terapool — APPROVED** [review]

File: src/constants.ts [hunk 5]
```diff
@@ -0,0 +1,10 @@
+const BACKEND_URL = 'https://lessoner.herokuapp.com';
+const EMAIL = {
+  min: 3,
+  max: 256,
+};
+const PASSWORD = {
+  min: 6,
```
Comment (NatashaMikhaltsova) [inline src/constants.ts:7]: I think, it will be better to rename constants to EMAIL_LENGTH and PASSWORD_LENGTH

**Review by NatashaMikhaltsova — CHANGES_REQUESTED** [review]

File: src/constants.ts (hunk 5 above)
Comment (Kurilo-Pavel) [reply src/constants.ts]: I rename keys constants EMAIL={minLength:6, maxLength=256} also PASSWORD

**Review by NatashaMikhaltsova — APPROVED** [review]

**Pull request merged**